src/Components/Homepage.js

```diff
@@ -1,38 +1,58 @@
-import React, {Component} from 'react';
+import React, { useState, useCallback, useEffect, useRef } from 'react';
+
 import Header from './Header';
 import Main from './Main';
 import Footer from './Footer';
 import ErrorBoundary from './ErrorBoundary';
-import './Homepage.css';
 import { MovieModal } from "./MovieModal";
+import { MovieInfo } from "./MovieInfo";
 
-class Homepage extends Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            isAddMovieModalShown: false
-        }
-    }
+import './Homepage.css';
+
+export const Homepage = () => {
+    const useToggle = (isComponentShown = false) => {
+        const [flag, setFlag] = useState(isComponentShown);
 
-    handleToggleAddMovieModal = () => {
-        this.setState(prevState => ({
-            isAddMovieModalShown: !prevState.isAddMovieModalShown
-        }))
+        const toggle = useCallback(() => {
+            setFlag(!flag)
+        }, [flag])
+        return [flag, toggle]
     }
 
-    render() {
-        return (
-            <div className='container'>
-                <Header title='find your movie' toggleAddMovieModal={this.handleToggleAddMovieModal}/>
-                <ErrorBoundary>
-                    {this.state.isAddMovieModalShown &&
-                    <MovieModal toggleMovieModal={this.handleToggleAddMovieModal} text={'add movie'}/>}
-                    <Main/>
-                </ErrorBoundary>
-                <Footer/>
-            </div>
+    const [movieInfo, setMovieInfo] = useState({});
+    const [isMovieInfoShown, setIsMovieInfoShown] = useState(false);
+    const [isAddMovieModalShown, toggleMovieModal] = useToggle(false);
+
+    const flag = useRef(true);
+
+    useEffect(() => {
+        if (flag.current) {
+            flag.current = false;
+        } else {
+            setIsMovieInfoShown(true)
+        }
+    }, [movieInfo]);
```

Comment:
I'm not sure about `flag` variable. Do you really need this?

Reply:
I did this because setIsMovieShown() is running after componetDidMount

Reply:
not sure that I understood you. It should be ok just like this:
```js
useEffect(() => {
        setIsMovieInfoShown(true)
}, [movieInfo]);
```

Reply:
I meant  that the ModalInfo is shown instead  of Header after first render. I want to see ModalInfo only after update movieInfo

Reply:
> I meant that the ModalInfo is shown instead of Header after first render

i believe that you can do it like this:
```js
useEffect(() => { 
  if (movieInfo) {
    setIsMovieInfoShown(true) 
  }
}, [movieInfo]);
```

but only thing that you need to initialize `movieInfo` with `null` instead of empty object
```js
const [movieInfo, setMovieInfo] = useState(null);
```

Reply:
It's working! Thanks! fixed